src/util/GestureHandler.as

```diff
@@ -358,11 +380,7 @@ public class GestureHandler {
 			}
 			o = o.parent;
 		}
-		var rect:Rectangle = app.stageObj().getRect(stage);
-		if(!mouseTarget && rect.contains(evt.stageX, evt.stageY))  return findMouseTargetOnStage(evt.stageX / app.scaleX, evt.stageY / app.scaleY);
-		if (o == null) return null;
 		if ((o is Block) && Block(o).isEmbeddedInProcHat()) return o.parent;
-		if (o is ScratchObj) return findMouseTargetOnStage(evt.stageX / app.scaleX, evt.stageY / app.scaleY);
```

Comment:
- The `if (!mouseTarget && rect.contains...` is now covered by the `checkScratchStage` logic above.
- So is the `if (o is ScratchObj)` case.
- The `if (o==null) return null` line is redundant given the `return o;` line below; it just short-circuits some `is` checking.